presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetViewModel.kt

```diff
@@ -1391,6 +1469,84 @@ class ObjectSetViewModel(
         }
     }
 
+    // region TEMPLATES
+    private suspend fun gettingTemplatesForDataViewState(state: ObjectState.DataView) {
+        when (state) {
+            is ObjectState.DataView.Collection -> {
+                proceedWithGettingTemplates(typeId = null)
+            }
+            is ObjectState.DataView.Set -> {
+                val sourceId = proceedWithGettingSetSourceId(state)
+                val sourceMap = state.details[sourceId] ?: return
+                when (sourceMap.type.firstOrNull()) {
+                    ObjectTypeIds.RELATION -> proceedWithGettingTemplates(typeId = null)
+                    ObjectTypeIds.OBJECT_TYPE -> proceedWithGettingTemplates(typeId = sourceId)
+                    else -> { Timber.d("Ignoring type of source") }
+                }
+            }
+        }
+    }
+
+    private suspend fun proceedWithGettingTemplates(typeId: Id?) {
+        val objectType = getObjectType(typeId)
+        if (objectType?.isTemplatesAllowed() == true) {
+            viewModelScope.launch {
+                getTemplates.execute(GetTemplates.Params(objectType.id)).fold(
```

Comment:
please check if it is not on main thread, or consider using async() method

Reply:
migrated to async